R/rules-other.R

```diff
@@ -1,17 +1,27 @@
-#' @importFrom rlang seq2
+#' @importFrom purrr reduce
 add_brackets_in_pipe <- function(pd) {
-  has_no_brackets <- (pd$token_before == "SPECIAL-PIPE") &
-    (pd$token == "SYMBOL") & (pd$text != ".")
-  if (!any(has_no_brackets)) return(pd)
-  new_pos_ids <- create_pos_ids(pd, 1, after = TRUE, n = 2)
-  new_pd <- create_tokens(
-    tokens = c("'('", "')'"), texts = c("(", ")"), pos_ids = new_pos_ids,
-    lag_newlines = rep(0, 2), parents = create_parent_id(pd))
-  pd <- bind_rows(pd, new_pd)
+  is_pipe <- pd$token == "SPECIAL-PIPE"
+  if (any(is_pipe, na.rm = TRUE)) {
+    pd <- reduce(which(is_pipe), add_brackets_in_pipe_one, .init = pd)
```

Comment:
Would the `reduce()` call also work if `which(is_pipe)` has length zero?

Reply:
Yes, thanks.